packages/now-routing-utils/test/index.spec.js

```diff
@@ -403,7 +411,82 @@ describe('normalizeRoutes', () => {
           },
           schemaPath: '#/items/additionalProperties',
         },
-      ],
+      ]
     );
   });
 });
+
+describe('getTransformedRoutes', () => {
+  test('should normalize nowConfig.routes', () => {
+    const nowConfig = { routes: [{ src: 'page', dest: '/page.html' }] };
+    const filePaths = [];
+    const actual = getTransformedRoutes({ nowConfig, filePaths });
+    const expected = normalizeRoutes(nowConfig.routes);
+    assert.deepEqual(actual, expected);
+    assertValid(actual.routes);
+  });
+
+  test('should normalize redirects before rewrites', () => {
+    const nowConfig = {
+      rewrites: [{ source: 'page', destination: '/page.html' }],
+      redirects: [{ source: 'v2', destination: '/api.py', statusCode: 302 }],
+    };
+    const filePaths = [];
+    const actual = getTransformedRoutes({ nowConfig, filePaths });
+    const expected = [
+      {
+        src: '^v2$',
+        headers: { Location: '/api.py' },
+        continue: true,
+        status: 302,
+      },
+      { src: '^page$', dest: '/page.html', continue: true },
+    ];
+    assert.deepEqual(actual.routes, expected);
+    assertValid(actual.routes);
+  });
+
+  test('should validate schemas', () => {
+    const nowConfig = {
+      cleanUrls: true,
+      rewrites: [
+        { source: 'page', destination: '/page.html' },
+        { source: 'home', destination: '/index.html' },
+      ],
+      redirects: [
+        { source: 'version1', destination: '/api1.py' },
+        { source: 'version2', destination: '/api2.py', statusCode: 302 },
+      ],
+      headers: [
+        {
+          source: '(.*)',
+          headers: [
+            {
+              key: 'Access-Control-Allow-Origin',
+              value: '*',
+            },
+          ],
+        },
+        {
+          source: '404.html',
```

Comment:
With clean URLs, does this match?

Reply:
Great, point. I guess it would match the redirect from `/404.html` to `/404` which is not what the user would intend. I'll change the source here to `/404`.